peppy/project.py

```diff
@@ -140,13 +139,13 @@ def __init__(
         self.st_index = (
             sample_table_index or getattr(self, "st_index", None) or SAMPLE_NAME_ATTR
         )
+
         self.sst_index = (
-            subsample_table_index
-            or getattr(self, "sst_index", None)
-            or [
-                SAMPLE_NAME_ATTR,
-                SUBSAMPLE_NAME_ATTR,
-            ]
+            ([subsample_table_index] if subsample_table_index else None)
+            or (
+                [sst_index] if (sst_index := getattr(self, "sst_index", None)) else None
```

Comment:
I believe walrus operators are only supported in python >= `3.8`. Just an FYI if peppy promises support for previous versions.